### PR TITLE
[Driver][SYCL] Address issue when unbundling for non-FPGA archive

### PR DESCRIPTION
**Pull request by mdtoguchi**
When unbundling a fat static archive and not targeting FPGA, the driver
would assert due to a mismatch of expected unbundled files within the
generated toolchain.  This was due to the generic nature in which we
were searching for AOCO type archives.

### CODE REVIEW

**Comment by mdtoguchi** [comment]
Failure doesn't look associated with my change.

File: clang/test/Driver/sycl-offload-static-lib.cpp [hunk 1]
```diff
@@ -150,3 +150,15 @@
...
+// RUN: touch %t.a
+// RUN: %clangxx -target x86_64-unknown-linux-gnu -fsycl %t.a -### 2>&1 \
+// RUN:  | FileCheck %s -check-prefixes=STATIC_ARCHIVE_NO_UNBUNDLE
+// RUN: %clangxx -target x86_64-unknown-linux-gnu -fsycl -fsycl-targets=spir64_gen-unknown-unknown-sycldevice %t.a -### 2>&1 \
+// RUN:  | FileCheck %s -check-prefixes=STATIC_ARCHIVE_NO_UNBUNDLE
+// RUN: %clangxx -target x86_64-unknown-linux-gnu -fsycl -fsycl-targets=spir64_fpga-unknown-unknown-sycldevice %t.a -### 2>&1 \
+// RUN:  | FileCheck %s -check-prefixes=STATIC_ARCHIVE_NO_UNBUNDLE
+// RUN: %clangxx -target x86_64-unknown-linux-gnu -fsycl -fsycl-targets=spir64_x86_64-unknown-unknown-sycldevice %t.a -### 2>&1 \
+// RUN:  | FileCheck %s -check-prefixes=STATIC_ARCHIVE_NO_UNBUNDLE
+// STATIC_ARCHIVE_NO_UNBUNDLE: clang-offload-bundler{{.*}}
```
Comment (AGindinson) [inline clang/test/Driver/sycl-offload-static-lib.cpp:164]: Could `STATIC_ARCHIVE_UNBUNDLE` or `STATIC_ARCHIVE_ALWAYS_UNBUNDLE` end up being a better check prefix? The 'negated' name really didn't parse for me until I figured it's named after the bug and not after the resolution.

Comment (mdtoguchi) [reply clang/test/Driver/sycl-offload-static-lib.cpp:164]: Yup, that can be updated to better reflect the check.

**Review by AGindinson — APPROVED** [review]
LGTM, thanks!

**Pull request merged**